src/assets/toolkit/scripts/lib/component/slideshow1.js

```diff
@@ -0,0 +1,134 @@
+'use strict';
+
+export class Slideshow {
+  constructor (element, {
+    slideHolder = element.querySelector('.js-Slideshow-slides'),
+    slides = element.querySelectorAll('.js-Slideshow-slide'),
+    nextTrigger = element.querySelector('.js-Slideshow-next'),
+    prevTrigger = element.querySelector('.js-Slideshow-prev'),
+    currentCount = element.querySelector('.js-Slideshow-current'),
+    totalCount = element.querySelector('.js-Slideshow-total'),
+    classForward = 'slide-forward',
+    classForwardReset = 'slide-forwardReset',
+    classBack = 'slide-back'
+  } = {}) {
+
+    this.slides = Array.from(slides);
+    this.numSlides = this.slides.length;
```

Comment:
You might consider defining [a "getter" method](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get) for this. So instead of defining it here, you'd have a method like...

```
get numSlides() {
  return this.slides.length;
}
```

Then you could just refer to `this.numSlides` and it'd always return that value. Magic!